assemblies/resources/bin/check_ports

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Licensed to The Apereo Foundation under one or more contributor license
+# agreements. See the NOTICE file distributed with this work for additional
+# information regarding copyright ownership.
+#
+# The Apereo Foundation licenses this file to you under the Educational
+# Community License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License
+# at:
+#
+#   http://opensource.org/licenses/ecl2.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# -----------------------------------------------------------------------
+# check if http or https port is available to run opencast otherwise exit
+
+PROGNAME=$(basename "$0")
+
+if [ "$PROGNAME" = "start-opencast" ]; then
+
+  if [ -d "$1" ]; then
+    SETENV_ETC="$1"
+  else
+    if [ -d "${KARAF_BASE}" ]; then
+      SETENV_ETC="${KARAF_BASE}/etc"
+    else
+      SETENV_ETC="etc"
+    fi
+  fi
+
+  check_port() {
+    a="$1"
+    if [ "$a" -eq "$a" ] 2>/dev/null ; then
+      port="$a"
+      hexPort=$(printf ":%04X " "$a")
+    else
+      b="${a##*:}"
+      port=$(eval echo "\$${b%%\}*}")
+      if [ "$port" -eq "$port" ] 2>/dev/null ; then
+        hexPort=$(printf ":%04X " "$port")
+      else
+        return 0;
+      fi
+    fi
+    if [ -d /proc/net ] ; then
+      for i in /proc/net/*; do
+        if  [ -f "$i" ] && [ -r "$i" ]; then
+          o=$(grep -c "$hexPort" "$i")
+          if [ "$o" = "1" ]; then
+            return 1;
+          fi
+        fi
+      done
+    else
+        if [ -x "$(command -v netstat)" ]; then
+          o=$(netstat -anp tcp | grep -i "listen" | grep -c -F ".$port ")
```

Comment:
`netstat -anp tcp` does not work on my system since the parameters do not all exist. But assuming that it would work, you would also want to use something like this instead of counting lines:

```sh
if netstat -anp tcp | grep -i "listen" | grep -q -F ".$port "; then
  return 1;
…
```

…there is no need to call an additional test since grep already does the test.

Reply:
Not sure if the arguments are macOS (or BSD) specific. The improved check works for me on macOS.

Reply:
these params were tested on macOS, as linux will use proc